pkg/install/install.go

```diff
@@ -152,67 +141,67 @@ func NewInstaller(ctx context.Context, log *logrus.Entry, _env env.Interface, db
 }
 
 func (i *Installer) AdminUpgrade(ctx context.Context) error {
-	steps := []interface{}{
-		action(i.initializeKubernetesClients),
-		action(i.startVMs),
-		condition{i.apiServersReady, 30 * time.Minute},
-		action(i.ensureBillingRecord), // belt and braces
-		action(i.fixLBProbes),
-		action(i.fixNSG),
-		action(i.fixPullSecret),
-		action(i.ensureGenevaLogging),
-		action(i.ensureIfReload),
-		action(i.ensureRouteFix),
-		action(i.upgradeCertificates),
-		action(i.configureAPIServerCertificate),
-		action(i.configureIngressCertificate),
-		action(i.upgradeCluster),
-		action(i.addResourceProviderVersion), // Run this last so we capture the resource provider only once the upgrade has been fully performed
+	steps := []steps.Step{
+		steps.Action(i.initializeKubernetesClients),
+		steps.Action(i.startVMs),
+		steps.Condition(i.apiServersReady, 30*time.Minute),
+		steps.Action(i.ensureBillingRecord), // belt and braces
+		steps.Action(i.fixLBProbes),
+		steps.Action(i.fixNSG),
+		steps.Action(i.fixPullSecret),
+		steps.Action(i.ensureGenevaLogging),
+		steps.Action(i.ensureIfReload),
+		steps.Action(i.ensureRouteFix),
+		steps.Action(i.upgradeCertificates),
+		steps.Action(i.configureAPIServerCertificate),
+		steps.Action(i.configureIngressCertificate),
+		steps.Action(i.upgradeCluster),
+		steps.Action(i.addResourceProviderVersion), // Run this last so we capture the resource provider only once the upgrade has been fully performed
 	}
 
 	return i.runSteps(ctx, steps)
 }
 
 // Install installs an ARO cluster
 func (i *Installer) Install(ctx context.Context, installConfig *installconfig.InstallConfig, platformCreds *installconfig.PlatformCreds, image *releaseimage.Image, bootstrapLoggingConfig *bootstraplogging.Config) error {
-	steps := map[api.InstallPhase][]interface{}{
+	steps := map[api.InstallPhase][]steps.Step{
 		api.InstallPhaseBootstrap: {
-			action(i.createDNS),
-			action(func(ctx context.Context) error {
+			steps.Action(i.createDNS),
+			steps.AuthorizationRefreshingAction(i.fpAuthorizer, steps.Action(func(ctx context.Context) error {
```

Comment:
nit: This will log confusing error message due to chaining :man_shrugging:  if we want to change anything. I'm ok as long as people know what they are reading in the logs
```
running step [AuthorizationRefreshingAction [Action github.com/Azure/ARO-RP/pkg/install.(*Installer).attachNSGsAndPatch-fm]]
steps.Run() running step [AuthorizationRefreshingAction [Action github.com/Azure/ARO-RP/pkg/install.(*Installer).deployResourceTemplate-fm]]
```